### PR TITLE
[BugFix] AUTO_INCREMENT column can be inserted by null using SELECT (#24795)

### LINKED ISSUE #24795: [BugFix] AUTO_INCREMENT column can be inserted by null using SELECT
Problem:
AUTO_INCREMENT column can not be specified by NULL through simple INSERT stmt. But It can be inserted by INSERT INTO t SELECT 1, NULL or INSERT INTO a t1 SELECT b from t2, where a is AUTO_INCREMENT column and b may contain NULL value.

Solution:
Check data in OlapTableSink::send_chunk, if AUTO_INCREMENT is specified in INSERT INTO SELECT stmt, the value can not be NULL.

### PR DESCRIPTION
**Pull request by srlch**
Fixes #24795

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 17 (64.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|11|17|64.71%|[229, 230, 312, 313, 331, 332]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 17 (64.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|11|17|64.71%|[229, 230, 312, 313, 331, 332]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 17 (64.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|11|17|64.71%|[229, 230, 312, 313, 331, 332]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 17 (64.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|11|17|64.71%|[234, 235, 317, 318, 336, 337]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 17 (64.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|11|17|64.71%|[234, 235, 317, 318, 336, 337]|

File: fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java [hunk 1]
```diff
@@ -222,10 +222,30 @@ public ExecPlan plan(InsertStmt insertStmt, ConnectContext session) {
                     Preconditions.checkState(!CollectionUtils.isEmpty(targetPartitionIds));
                     enableAutomaticPartition = olapTable.supportedAutomaticPartition();
                 }
+                boolean nullExprInAutoIncrement = false;
+                // In INSERT INTO SELECT, if AUTO_INCREMENT column
+                // is specified, the column values must not be NULL
+                if (!(queryRelation instanceof ValuesRelation)) {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java:228]: Does the CTAS or materialized view also need to modify?

Comment (srlch) [reply fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java]: CTAS need it and materialized view maybe not, I have update the code

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 20 / 22 (90.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|20|22|90.91%|[239, 240]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 17 (64.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|11|17|64.71%|[239, 240, 322, 323, 341, 342]|

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**